patroni/__init__.py

```diff
@@ -145,3 +141,42 @@ def main():
         else:
             patroni.ha.while_not_sync_standby(lambda: patroni.postgresql.stop(checkpoint=False))
             patroni.dcs.delete_leader()
+
+
+def main():
+    if os.getpid() != 1:
+        return patroni_main()
+
+    pid = 0
+
+    # Looks like we are in a docker, so we will act like init
+    def sigchld_handler(signo, stack_frame):
+        try:
+            while True:
+                ret = os.waitpid(-1, os.WNOHANG)
+                if ret == (0, 0):
+                    break
+                elif ret[0] != pid:
+                    logging.info('Reaped unknow pid=%s, exit status=%s', *ret)
+        except OSError:
+            pass
+
+    def passtochild(signo, stack_frame):
+        if pid:
+            os.kill(pid, signo)
+
+        if signo in (signal.SIGTERM, signal.SIGINT, signal.SIGQUIT):
+            sys.exit(0)
```

Comment:
At least for SIGINT it would make sense to pass it on. Not an expert on containers, but I would guess that otherwise the child processes might get terminated before they can execute a graceful shutdown. Looking at a [project that has a similar goal](https://github.com/Yelp/dumb-init/blob/master/dumb-init.c#L90) doing something about SIGTSTP, SIGTTOU, SIGTTIN might or might not make sense here. Otherwise the code seems to match what dumb-init does.